src/test/java/com/springboot/testrunner/httpclient/ProjectApacheHttpClient.java

```diff
@@ -0,0 +1,224 @@
+                requestBuilder.addHeader((String) key, (String) headersMap.get(key));
+                LOGGER.info("Overrode header key:{}, with value:{}", key, headersMap.get(key));
+            }
+
+            isFileUpload = ((String) headersMap.get("content-type")).contains("multipart/form-data");
+
+        }
+
+        /** -----------------------------------------------------------------------------------
+         * Setting cookies:
+         *
+         * Highly discouraged to use sessions, but in case of any server dependent upon session,
+         * then it's taken care here.
+         * ------------------------------------------------------------------------------------
+         */
+        if (COOKIE_JSESSIONID_VALUE != null) {
+            requestBuilder.addHeader("Cookie", (String) COOKIE_JSESSIONID_VALUE);
+        }
+
+        // =-=-=-=-=-=-=-=-=-=-=-=-
+        // Now execute the request
+        // =-=-=-=-=-=-=-=-=-=-=-=-
+        CloseableHttpResponse httpResponse;
+        ///
+        if(isFileUpload){
+            LOGGER.info("Zerocode - Preparing file upload");
+
+            RequestBuilder uploadRequestBuilder = createFileUploadRequestBuilder(httpUrl, methodName, reqBodyAsString);
+
+            LOGGER.info("Zerocode - Executing file upload");
+
+            httpResponse = httpclient.execute(uploadRequestBuilder.build());
+
+        } else {
+
+            httpResponse = httpclient.execute(requestBuilder.build());
+        }
+
+        HttpEntity entity = httpResponse.getEntity();
+        Response serverResponse = Response
+                .status(httpResponse.getStatusLine().getStatusCode())
+                .entity(entity != null ? IOUtils.toString(entity.getContent()) : null)
+                .build();
+
+
+        Header[] allHeaders = httpResponse.getAllHeaders();
+        Response.ResponseBuilder responseBuilder = Response.fromResponse(serverResponse);
+        for (Header thisHeader : allHeaders) {
+            String headerKey = thisHeader.getName();
+            responseBuilder = responseBuilder.header(headerKey, thisHeader.getValue());
+
+            /** ---------------
+             * Session handled
+             * ----------------
+             */
+            if ("Set-Cookie".equals(headerKey)) {
+                COOKIE_JSESSIONID_VALUE = serverResponse.getMetadata().get(headerKey);
+            }
+        }
+        serverResponse = responseBuilder.build();
+
+        return serverResponse;
+    }
+
+    private RequestBuilder createFileUploadRequestBuilder(String httpUrl, String methodName, String reqBodyAsString) throws IOException {
+        String fileName = getFileNameFromPayLoad(reqBodyAsString);
+        FileBody file = new FileBody(new File(getAbsPath(fileName)));
+
+        HttpEntity reqEntity = MultipartEntityBuilder.create()
+                .addPart("file", file)
+                .setBoundary("" + System.currentTimeMillis() + LocalDateTime.now())
```

Comment:
`setBoundary(...)` argument is good to **default** it to `current time`, but in this example allow it to pick from the request, if not present, then default it to current time(which is done already)